apps/rush-lib/src/utilities/VersionControl.ts

```diff
@@ -48,49 +55,49 @@ export class VersionControl {
    * @param repositoryUrl - repository url
    */
   public static getRemoteMasterBranch(repositoryUrl?: string): string {
-    const defaultRemote: string = 'origin';
-    const defaultMaster: string = 'origin/master';
-    let useDefault: boolean = false;
     let matchingRemotes: string[] = [];
 
-    if (!repositoryUrl) {
-      useDefault = true;
-    } else {
+    if (repositoryUrl) {
       const output: string = child_process
-      .execSync(`git remote`)
-      .toString();
+        .execSync(`git remote`)
+        .toString();
       matchingRemotes = output.split('\n').filter(remoteName => {
         if (remoteName) {
           const remoteUrl: string = child_process.execSync(`git remote get-url ${remoteName}`)
             .toString()
             .trim();
-          if (remoteName === defaultRemote && remoteUrl === repositoryUrl) {
-            useDefault = true;
-          }
           return remoteUrl === repositoryUrl;
         }
         return false;
       });
+    } else {
+      console.log(colors.yellow(
+        'A remote URL has not been specified in rush.json. Setting the baseline remote URL is recommended.'
+      ));
+      return DEFAULT_FULLY_QUALIFIED_BRANCH;
     }
 
-    if (useDefault) {
-      return defaultMaster;
-    } else if (matchingRemotes.length > 0) {
+    if (matchingRemotes.length > 0) {
       if (matchingRemotes.length > 1) {
-        console.log(`More than one remotes match the repository url. Use the first remote.`);
+        console.log(`More than one remote matches the repository URL. Using the first remote (${matchingRemotes[0]}).`);
```

Comment:
In these error messages, perhaps we should say "Git remote URL" instead of "remote URL".  The word "remote" is pretty opaque jargon and many people may have no  idea what it's referring to.